app/Repositories/BaseRepository.php

```diff
@@ -0,0 +1,343 @@
+    public function create(array $data)
+    {
+        return $this->model->create($data);
+    }
+
+    /**
+     * @param  array  $data      Array data
+     * @param  $id ID
+     * @param  string $attribute Attribute
+     * @return void
+     */
+    public function update(array $data, $id, $attribute = "id")
+    {
+         return $this->model->where($attribute, '=', $id)->update($data);
+    }
+
+    /**
+     * @param  array      $data      Array data
+     * @param  $id ROLE_ID
+     * @param  string     $attribute Attribute
+     * @return void
+     */
+    public function update_role(array $data, $role_id, $attribute = "role_id")
+    {
+         return $this->model->where($attribute, '=', $role_id)->update($data);
+    }
+
+    /**
+     * @param  $id ID
+     * @return void
+     */
+    public function delete($id)
+    {
+        return $this->model->destroy($id);
+    }
+
+    /**
+     * @param  $id ID
+     * @param  array $columns Columns
+     * @return array Return array
+     */
+    public function find($id, $columns = ['*'])
+    {
+        return $this->model->find($id, $columns);
+    }
+
+    /**
+     * @param  $attribute Attribute
+     * @param  $value Value
+     * @param  array               $columns Columns
+     * @return array Array
+     */
+    public function findBy($attribute, $value, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->first($columns);
+    }
+
+    public function findByReturnArray($attribute, $value, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->get($columns);
+    }
+
+    public function findByReturnArrayPlug($attribute, $value, $pluck, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->get($columns)->pluck($pluck);
+    }
+    public function findByMultiConditionsLimit($whereData = [], $columns = ['*'])
+    {
+        return $this->model->where($whereData)->first($columns);
+    }
+    /**
+     * @param  $tableName Table name
+     * @param  array                $whereData Array where data
+     * @return array Return array
+     */
+    public function findByMultiConditions($tableName, $whereData = [])
+    {
+        return DB::table($tableName)->where($whereData)->get();
+    }
+
+    /**
+     * @param  $tableName Table name
+     * @param  array                $whereData Array where data
+     * @return array Return array
+     */
+    public function pagingWithMultiConditions($tableName, $whereData = [], $perPage = 15, $orderBy = 'id desc', $columns = ['*'])
+    {
+        return DB::table($tableName)
+            ->where($whereData)
+            ->orderByRaw($orderBy)
+            ->paginate($perPage, $columns);
+    }
+
+    public function delByMultiConditions($tableName, $whereData = [])
+    {
+        return DB::table($tableName)
+            ->where($whereData)
+            ->delete();
+    }
+
+    /**
+     * @param  array $whereData
+     * @return mixed
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/BaseRepository.php</br>**Line**: 272</br>String "*" does not require double quotes; use single quotes instead